### DISCUSSION TITLE
Best Way to Screenshot a Markdown File?

### DISCUSSION THREAD

**Discussion by Mike-E-angelo** [hardkoded/puppeteer-sharp#2696] [Q&A] [answered]
Hello Community,

I am using PupeteerSharp to take screenshots of files loaded in a Chrome browser.  Everything is working pretty darn well except for Markdown files.  It does not appear that Markdown is natively allowed in Chrome and an extension is necessary.

I did look into installing extensions but there is a complication here where the file being loaded is on the local machine (served out of the service account's `%AppData%/Temp` directory) and special permission is required to do this.  I was not able to figure this out so I am writing here to see:

1. Is it possible to set this permission (`Allow access to file URLs`) in PupeteerSharp via C# (and if so, how)?
2. Is there an easier way of loading Markdown files via PupeteerSharp that I should be aware of?  If there is some way of doing this without an extension this would be most ideal.

Thank you for any assistance you can provide.

**Comment by kblok** [comment, marked as answer]
You could generate the HTML yourself using something like [this](https://github.com/showdownjs/showdown) and then navigate to that html

**Reply by Mike-E-angelo** [reply]
Wow, thank you @kblok that is a highly valuable but _painfully_ obvious suggestion to me. 😅  I am disappointed in myself that I didn't think of that.  I appreciate your assistance in getting me pointed in the right direction!
